src/internal/drawCompositeImage.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * This module is responsible for drawing an image to an enabled elements canvas element
+ */
+
+(function ($, cornerstone) {
+
+    "use strict";
+
+    // This is used to keep each of the layers' viewports in sync with the active layer
+    var syncedViewports = {};
+
+    // Create a copy of the properties that will be cached when syncing viewports
+    function cloneViewport(viewport) {
+        return {
+            rotation: viewport.rotation,
+            scale: viewport.scale,
+            translation: {
+                x: viewport.translation.x,
+                y: viewport.translation.y
+            }
+        }
+    }
+
+    function getDrawImageOffset(targetImageId, referenceImageId) {
+        var offset = {
+            x: 0,
+            y: 0
+        };
+
+        var targetImagePlane = cornerstone.metaData.get('imagePlane', targetImageId);
+        if (!targetImagePlane ||
+            !targetImagePlane.imagePositionPatient ||
+            !targetImagePlane.rowCosines ||
+            !targetImagePlane.columnCosines) {
+            return offset;
+        }
+
+        var referenceImagePlane = cornerstone.metaData.get('imagePlane', referenceImageId);
+        if (!referenceImagePlane ||
+            !referenceImagePlane.imagePositionPatient ||
+            !referenceImagePlane.rowCosines ||
+            !referenceImagePlane.columnCosines) {
+            return offset;
+        }
+
+        // TODO: Add Image Orientation check between layers
+        var pos = targetImagePlane.imagePositionPatient;
+        var origPos = referenceImagePlane.imagePositionPatient
+
+        // UNCOMMENT THESE LINES AFTER FIXING IT
```

Comment:
I need to confirm with Erik because the layer seems to be well aligned after updating its zoom (scale) based on pixel spacing * width (see `rescaleImage` below).